src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerBuilder.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.util.Optional;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Builds a {@link CircuitBreaker} instance using builder pattern.
+ */
+public final class CircuitBreakerBuilder {
+
+    private static final class Defaults {
+
+        private static final double FAILURE_RATE_THRESHOLD = 0.8;
+
+        private static final ExceptionFilter EXCEPTION_FILTER = cause -> true;
+
+        private static final long MINIMUM_REQUEST_THRESHOLD = 10;
+
+        private static final Duration TRIAL_REQUEST_INTERVAL = Duration.ofSeconds(3);
+
+        private static final Duration CIRCUIT_OPEN_WINDOW = Duration.ofSeconds(10);
+
+        private static final Duration COUNTER_SLIDING_WINDOW = Duration.ofSeconds(20);
+
+        private static final Duration COUNTER_UPDATE_INTERVAL = Duration.ofSeconds(1);
+
+        private static final Clock CLOCK = Clock.systemUTC();
+    }
+
+    private final Optional<String> name;
+
+    private double failureRateThreshold = Defaults.FAILURE_RATE_THRESHOLD;
+
+    private ExceptionFilter exceptionFilter = Defaults.EXCEPTION_FILTER;
+
+    private long minimumRequestThreshold = Defaults.MINIMUM_REQUEST_THRESHOLD;
+
+    private Duration trialRequestInterval = Defaults.TRIAL_REQUEST_INTERVAL;
+
+    private Duration circuitOpenWindow = Defaults.CIRCUIT_OPEN_WINDOW;
+
+    private Duration counterSlidingWindow = Defaults.COUNTER_SLIDING_WINDOW;
+
+    private Duration counterUpdateInterval = Defaults.COUNTER_UPDATE_INTERVAL;
+
+    private Clock clock = Defaults.CLOCK;
```

Comment:
I've just realized `Clock` uses `System.currentTimeMillis()` internally. Could we use `Ticker` instead?

https://github.com/line/armeria/blob/b2b5a23c2ae56af881dca171b06d9a063f624bed/src/main/java/com/linecorp/armeria/common/util/Ticker.java